services/intersection-api/api/src/test/java/us/dot/its/jpo/ode/api/accessorTests/assessments/ConnectionOfTravelAssessmentRepositoryImplTest.java

```diff
@@ -55,133 +55,148 @@
+        aggregationResultCount.setCount(1L);
+        aggregationResult.setMetadata(List.of(aggregationResultCount));
+
+        when(mockAggregationResult.getUniqueMappedResult()).thenReturn(aggregationResult);
+
+        ArgumentCaptor<Aggregation> aggregationCaptor = ArgumentCaptor.forClass(Aggregation.class);
+        doReturn(mockAggregationResult).when(mongoTemplate).aggregate(aggregationCaptor.capture(),
+                anyString(),
+                eq(AggregationResult.class));
+
+        // Call the repository find method
+        PageRequest pageRequest = PageRequest.of(0, 1);
+        Page<ConnectionOfTravelAssessment> findResponse = repository.find(intersectionID, startTime, endTime,
+                pageRequest);
+
+        verify(mongoTemplate).aggregate(any(Aggregation.class), anyString(),
+                eq(AggregationResult.class));
+
+        // Extract the captured Aggregation
+        Aggregation capturedAggregation = aggregationCaptor.getValue();
+
+        // Extract the MatchOperation from the Aggregation pipeline
+        Document pipeline = capturedAggregation.toPipeline(Aggregation.DEFAULT_CONTEXT).get(0);
+
+        // Assert the Match operation Criteria
+        assertThat(pipeline.toJson())
+                .isEqualTo(String.format(
+                        "{\"$match\": {\"intersectionID\": %s, \"assessmentGeneratedAt\": {\"$gte\": {\"$date\": \"%s\"}, \"$lte\": {\"$date\": \"%s\"}}}}",
+                        intersectionID, startTimeString, endTimeString));
+
+        // Serialize results to JSON and compare with the original JSON
+        String resultJson = objectMapper.writeValueAsString(findResponse.getContent().get(0));
+
+        // Remove unused fields from each entry
+        List<Document> expectedResult = sampleDocuments.stream().map(doc -> {
+            doc.remove("_id");
+            doc.remove("recordGeneratedAt");
+            return doc;
+        }).toList();
+        String expectedJson = objectMapper.writeValueAsString(expectedResult.get(0));
+
+        // Compare JSON with ignored fields
+        JSONAssert.assertEquals(expectedJson, resultJson, new CustomComparator(
+                JSONCompareMode.LENIENT, // Allows different key orders
+                new Customization("properties.timeStamp", (_, _) -> true),
+                new Customization("properties.odeReceivedAt", (_, _) -> true)));
+    }
+
+    @Test
+    void testFindLatest() {
+        ConnectionOfTravelAssessment event = new ConnectionOfTravelAssessment();
+        event.setIntersectionID(intersectionID);
+
+        doReturn(event).when(mongoTemplate).findOne(any(Query.class), eq(ConnectionOfTravelAssessment.class),
+                anyString());
+
+        Page<ConnectionOfTravelAssessment> page = repository.findLatest(intersectionID, startTime, endTime);
+
+        assertThat(page.getContent()).hasSize(1);
+        assertThat(page.getContent().get(0).getIntersectionID()).isEqualTo(intersectionID);
```

Comment:
suggestion(non-blocking): use the built-in method to get the first item in a collection
```suggestion
        assertThat(page.getContent().getFirst().getIntersectionID()).isEqualTo(intersectionID);
```

Reply:
there are 165 cases of this, I will replace a few that look like this but leave the rest the same!